core/src/main/scala/codesearch/core/regex/lexer/Tokenizer.scala

```diff
@@ -13,8 +13,9 @@ object Tokenizer {
   /** A POSIX character class, e.g. `[:alpha:]`. */
   private def charSetPred[_: P] = P("[:" ~ (!":]" ~ AnyChar).rep ~ ":]")
 
+  /**Handles cases {n} and {n,} and {n, n}, example {33,}*/
   private def parseRepetitionSeq[_: P] =
-    P("{" ~ (CharIn("0-9").rep ~ ("," ~ CharIn("0-9").rep | "," | "")) ~ "}").!.map(RepetitionSeq(_))
+    P("{" ~ (CharIn("0-9").rep(1) ~ ("," ~ CharIn("0-9").rep | "")) ~ "}").!.map(RepetitionSeq(_))
 
   private def specialSymbols[_: P] =
```

Comment:
Let's move `Space` into a separate variant, it would be cleaner this way.

Can be done in a separate PR, later.

Reply:
Do not quite understand. Please explain.

Reply:
@Evengining Please arrange functions in the order they are called for better readability Also public functions must be upper then private.